pkg/pillar/cmd/volumemgr/handlediskmetrics.go

```diff
@@ -0,0 +1,276 @@
+	}
+}
+
+func unpublishDiskMetricStatuses(ctx *volumemgrContext, statuses ...*types.DiskMetricStatus) {
+	for _, status := range statuses {
+		key := status.Key()
+		log.Debugf("unpublishDiskMetricStatuses(%s)", key)
+		pub := ctx.pubDiskMetricStatus
+		c, _ := pub.Get(key)
+		if c == nil {
+			log.Errorf("unpublishDiskMetricStatuses(%s) not found", key)
+			continue
+		}
+		pub.Unpublish(key)
+		log.Debugf("unpublishDiskMetricStatuses(%s) Done", key)
+	}
+}
+
+func lookupDiskMetricStatus(ctx *volumemgrContext, key string) *types.DiskMetricStatus {
+
+	log.Debugf("lookupDiskMetricStatus(%s)", key)
+	pub := ctx.pubDiskMetricStatus
+	c, _ := pub.Get(key)
+	if c == nil {
+		log.Debugf("lookupDiskMetricStatus(%s) not found", key)
+		return nil
+	}
+	status := c.(types.DiskMetricStatus)
+	log.Debugf("lookupDiskMetricStatus(%s) Done", key)
+	return &status
+}
+
+func lookupDiskMetricStatusByDiskPath(ctx *volumemgrContext, diskPath string) *types.DiskMetricStatus {
+	log.Debugf("lookupDiskMetricStatusByDiskPath(%s)", diskPath)
+	pub := ctx.pubDiskMetricStatus
+	items := pub.GetAll()
+	for _, st := range items {
+		status := st.(types.DiskMetricStatus)
+		if status.DiskPath == diskPath || status.MountPath == diskPath {
+			log.Debugf("lookupDiskMetricStatusByDiskPath(%s) Done", diskPath)
+			return &status
+		}
+	}
+	log.Debugf("lookupDiskMetricStatusByDiskPath(%s) not found", diskPath)
+	return nil
+}
+
+func publishAppDiskMetricStatuses(ctx *volumemgrContext, statuses ...*types.AppDiskMetricStatus) {
+	for _, status := range statuses {
+		key := status.Key()
+		log.Debugf("publishAppDiskMetricStatuses(%s)", key)
+		pub := ctx.pubAppDiskMetricStatus
+		pub.Publish(key, *status)
+		log.Debugf("publishAppDiskMetricStatuses(%s) Done", key)
+	}
+}
+
+func unpublishAppDiskMetricStatuses(ctx *volumemgrContext, statuses ...*types.AppDiskMetricStatus) {
+	for _, status := range statuses {
+		key := status.Key()
+		log.Debugf("unpublishAppDiskMetricStatuses(%s)", key)
+		pub := ctx.pubAppDiskMetricStatus
+		c, _ := pub.Get(key)
+		if c == nil {
+			log.Errorf("unpublishAppDiskMetricStatuses(%s) not found", key)
+			continue
+		}
+		pub.Unpublish(key)
+		log.Debugf("unpublishAppDiskMetricStatuses(%s) Done", key)
+	}
+}
+
+func lookupAppDiskMetricStatus(ctx *volumemgrContext, key string) *types.AppDiskMetricStatus {
+
+	log.Debugf("lookupAppDiskMetricStatus(%s)", key)
+	pub := ctx.pubAppDiskMetricStatus
+	c, _ := pub.Get(key)
+	if c == nil {
+		log.Debugf("lookupAppDiskMetricStatus(%s) not found", key)
+		return nil
+	}
+	status := c.(types.AppDiskMetricStatus)
+	log.Debugf("lookupAppDiskMetricStatus(%s) Done", key)
+	return &status
+}
+
+//diskMetricsTimerTask calculates and publishes disk metrics periodically
+func diskMetricsTimerTask(ctx *volumemgrContext) {
+	log.Infoln("starting report diskMetricsTimerTask timer task")
+	createOrUpdateDiskMetrics(ctx)
+
+	diskMetricInterval := time.Duration(300) * time.Second
```

Comment:
How about making this a ConfigItem so it can be set from the controller?

Reply:
Done.